### PR TITLE
Enable env var hints for `--token`

### PR DESCRIPTION
**Pull request by andersjanmyr**
Example:

```
$ ./fastly list service --token $FASTLY<tab>
$FASTLY_ACCOUNT_DEV         $FASTLY_ACCOUNT_SALJA_PROD
$FASTLY_ACCOUNT_PROD        $FASTLY_ACCOUNT_SANDBOX
$FASTLY_ACCOUNT_SALJA_DEV
```

### CODE REVIEW

**Review by Integralist — CHANGES_REQUESTED** [review]
👋🏻 @andersjanmyr thanks for opening this PR! 

Implementation looks OK, I just have a few requests.

File: pkg/app/run.go [hunk 1]
```diff
@@ -30,6 +30,20 @@ func FastlyAPIClient(token, endpoint string) (api.Interface, error) {
 	return client, err
 }
 
+var envVarCache []string
```
[the remark this answers is not in the document]
Comment (andersjanmyr) [reply pkg/app/run.go:33]: Yes, makes sense. I was thinking it was the same invokation every time I press tab, but it's a new every time.

File: pkg/app/run.go [hunk 2]
```diff
@@ -30,6 +30,20 @@ func FastlyAPIClient(token, endpoint string) (api.Interface, error) {
...
+
+func envVars() []string {
+	if envVarCache != nil {
+		return envVarCache
+	}
+	envVarCache = []string{}
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		envVarCache = append(envVarCache, fmt.Sprintf("\\$%s", pair[0]))
```
[the remark this answers is not in the document]
Comment (andersjanmyr) [reply pkg/app/run.go:42]: Yes, it probably needs some special treatment to handle Windows.

**Comment by Integralist** [comment]
@andersjanmyr out of interest, is there a reason you're using this approach rather than something like `--profile` (e.g. `fastly service list --profile personal`)?

**Comment by andersjanmyr** [comment]
@Integralist I was thinking of using profile locally, but we're using the same script to deploy both locally and on CI.
Something like this.
```
case "$env" in
  prod|stage)
    export FASTLY_API_KEY=$FASTLY_ACCOUNT_PROD
    ;;
  main)
    env='dev'
    export FASTLY_API_KEY=$FASTLY_ACCOUNT_DEV
    ;;
  *)
    export FASTLY_API_KEY=$FASTLY_ACCOUNT_DEV
    ;;
esac
```

File: pkg/env/env.go [hunk 3]
```diff
@@ -17,3 +24,27 @@ const (
 	// CustomerID is the env var we look in for a Customer ID.
 	CustomerID = "FASTLY_CUSTOMER_ID"
 )
+
+func Vars() []string {
```
Comment (Integralist) [inline pkg/env/env.go:28]: We should annotate this public function.

File: pkg/env/env.go [hunk 4]
```diff
@@ -17,3 +24,27 @@ const (
...
+	vars := []string{}
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		vars = append(vars, toVar(pair[0]))
+	}
+	return vars
+}
+
+func toVar(v string) string {
+	if runtime.GOOS == "windows" {
```
Comment (Integralist) [inline pkg/env/env.go:38]: For consistency could we replace this with https://github.com/fastly/cli/blob/main/pkg/runtime/runtime.go#L11

**Review by Integralist — CHANGES_REQUESTED** [review]
Other than two minor change requests this looks good to me.

Thanks!

**Pull request merged**